src/prefect_server/graphql/extensions.py

```diff
@@ -36,6 +37,22 @@ def log_error(exc: Exception) -> None:
         logger.error({"traceback": traceback.format_exc(), "context": ctx})
 
 
+class PrefectConcurrencyLimiter(Extension):
```

Comment:
A single process global concurrency lock. I'm not married to this implementation (or naming), but it seemed to fit well with the injection of the `headers` into the context in the `PrefectHeader` extension.